web3.js/src/util/send-and-confirm-transaction.ts

```diff
@@ -33,17 +39,81 @@ export async function sendAndConfirmTransaction(
     sendOptions,
   );
 
-  const status = (
-    await connection.confirmTransaction(
-      signature,
-      options && options.commitment,
-    )
-  ).value;
+  const subscriptionCommitment =
+    options?.preflightCommitment || options?.commitment;
+  let subscriptionId: number | undefined;
+  let status: RpcResponseAndContext<SignatureResult> | null = null;
+  let message: string | null = null;
+
+  const confirmTx = new Promise((resolve, reject) => {
+    try {
+      subscriptionId = connection.onSignature(
+        signature,
+        (result: SignatureResult, context: Context) => {
+          subscriptionId = undefined;
+          status = {
```

Comment:
You never actually use this except to check if it's non-null. Is there a reason why we can't just eliminate `status` and have a `succeeded` flag instead?

Reply:
Okay, after looking into this more I think I understand how to use this appropriately. I kept it as is but moved it back into `confirmTransaction` and changed `status` back to `response`. Then I check for `err` (which is on `SignatureResult`) like in the original implementation.